### ISSUE TITLE
Hermes is always disabled

### ISSUE THREAD

**Issue by DanielSRS** [microsoft/react-native-macos#2288]
### Environment

```markdown
react-native -v: 0.75.4
npm ls react-native-macos: 0.75.15
node -v: 21.7.1
npm -v: 10.5.0
yarn --version: 4.5.1
xcodebuild -version: 15.4
```

### Steps to reproduce the bug

Not exactly sure if this is a bug or intended, But I can't enable Hermes by just running `USE_HERMES=1 pod install` [as suggested here](https://github.com/microsoft/react-native-macos/issues/1859#issuecomment-1605162902) (since the docs site seems outdated) because in Podfile:

```Podfile
:hermes_enabled => false
```

I'm unaware if there is any problem with Hermes on MacOS that motivated this, I was just testing DevTools and Hermes is needed for that.

### Expected Behavior

just run `USE_HERMES=1 pod install` to enable Hermes

### Actual Behavior

running `USE_HERMES=1 pod install` does nothing

### Reproducible Demo

To reproduce:

- run `npx react-native-macos-init` to add macos to a RN project
- run cd macos && `USE_HERMES=1 pod install` 

### Additional context

I just updated the Podfile to:

```diff
use_react_native!(
    :path => '../node_modules/react-native-macos',
-   :hermes_enabled => false
+   :hermes_enabled => ENV['USE_HERMES'] == '1',
    :fabric_enabled => ENV['RCT_NEW_ARCH_ENABLED'] == '1',
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )
```

And all the related pods seems to be installed correctly and build and running worked without any issues so far.

If the template is intended to be like this, feel free to close it right away.

**Comment by bear-ei** [comment]
In version 0.75.0, you can temporarily modify `:hermes_enabled => false` to `:hermes_enabled => true`, and then run `pod install` to enable Hermes.

**Comment by DanielSRS** [comment]
Yes, you're right, I was also wondering if it might be worth updating the template to simply check the env.

I think this would be more suitable to open as an discussion rather than a issue, but its not available in this repo.

**Comment by bear-ei** [comment]
> Yes, you're right, I was also wondering if it might be worth updating the template to simply check the env.
> 
> I think this would be more suitable to open as an discussion rather than a issue, but its not available in this repo.

I believe that using `USE_HERMES=1 pod install` is the correct approach. However, for versions 0.75.0 to 0.76.0 at least, if you do not use `:hermes_enabled => true`, then every time you need to update the pods, you must add the `USE_HERMES=1` instruction. It seems that the environment variable does not properly persist the `USE_HERMES` setting. Currently, using `:hermes_enabled => true` should only be a temporary measure.

**Comment by Saadnajmi** [comment]
Hermes is now the default.
